File: 007 - 10001st prime.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _10001prime
{
    internal class Program
    {
        static bool isPrime(int num)
        {
            for (int i = 2; i < (int)Math.Sqrt(num)+1; i++)
            {
                if (num % i == 0)
                {
                    return false;
                }
            }
            return true;
        }
        static void Main(string[] args)
        {
            int primes = 0;
            int current = 1;
            while (primes < 10001)
            {
                current++;
                if (isPrime(current))
                {
                    primes++;
                }
            }
            Console.WriteLine(current);
            Console.ReadKey();
        }
    }
}

File: 007 - 10001st prime.py
from math import isqrt
from itertools import count

def isPrime(n):
    for i in range(2, isqrt(n)+1):
        if n % i == 0:
            return False
    return True

c = 0
for i in count(2):
    if isPrime(i):
        c += 1
        if c == 10001:
            print(i)
            break